From 1edfda70ce60ddd4bb0cca2e2197d0c05009c916 Mon Sep 17 00:00:00 2001
From: Hardikkumar Panchal <hpanchal@qti.qualcomm.com>
Date: Mon, 13 Apr 2026 15:00:30 +0530
Subject: [PATCH] netfilter: conntrack: Use system_unbound_wq workqueue for
 conntrack_gc_work

Using system_power_efficient_wq makes conntrack worker bound to specific core,
This leads to starvation of this thread when CPU is fully used by softirq.

Change-Id: I520df9def0777b0f7dd821ac59cf99921fbf2157
Signed-off-by: Hardikkumar Panchal <hpanchal@qti.qualcomm.com>
---
 net/netfilter/nf_conntrack_core.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/net/netfilter/nf_conntrack_core.c b/net/netfilter/nf_conntrack_core.c
index 152a995eb6c9..f88fd8a4199d 100644
--- a/net/netfilter/nf_conntrack_core.c
+++ b/net/netfilter/nf_conntrack_core.c
@@ -1613,7 +1613,7 @@ static void gc_worker(struct work_struct *work)
 	if (next_run)
 		gc_work->early_drop = false;
 
-	queue_delayed_work(system_power_efficient_wq, &gc_work->dwork, next_run);
+	queue_delayed_work(system_unbound_wq, &gc_work->dwork, next_run);
 }
 
 static void conntrack_gc_work_init(struct conntrack_gc_work *gc_work)
@@ -2753,7 +2753,7 @@ int nf_conntrack_init_start(void)
 		goto err_proto;
 
 	conntrack_gc_work_init(&conntrack_gc_work);
-	queue_delayed_work(system_power_efficient_wq, &conntrack_gc_work.dwork, HZ);
+	queue_delayed_work(system_unbound_wq, &conntrack_gc_work.dwork, HZ);
 
 	ret = register_nf_conntrack_bpf();
 	if (ret < 0)
-- 
2.34.1

